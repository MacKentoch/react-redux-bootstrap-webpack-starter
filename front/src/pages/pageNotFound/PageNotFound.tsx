import React from 'react';
import { RouteComponentProps, withRouter } from 'react-router';
import Jumbotron from 'reactstrap/lib/Jumbotron';
import FadeInEntrance from '../../components/fadeInEntrance';
import { OwnProps, ReduxConnectedProps } from './index';

// #region types
export type Props = RouteComponentProps & ReduxConnectedProps & OwnProps;
// #endregion

function PageNotFound() {
  return (
    <FadeInEntrance>
      <Jumbotron>
        <h1>Sorry this page does not exists...</h1>
      </Jumbotron>
    </FadeInEntrance>
  );
}

PageNotFound.displayName = 'PageNotFound';

export default withRouter(PageNotFound);
