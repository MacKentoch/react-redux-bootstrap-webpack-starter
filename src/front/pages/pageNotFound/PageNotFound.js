// @flow

// #region lifecycle
import React, { PureComponent } from 'react';
import {
  type Match,
  type Location,
  type RouterHistory,
} from 'react-router-dom';
import Jumbotron from 'reactstrap/lib/Jumbotron';
// #endregion

// #region flow types
type Props = {
  // react-router 4:
  match: Match,
  location: Location,
  history: RouterHistory,

  ...any,
};
type State = any;
// #endregion

class PageNotFound extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    return (
      <div>
        <Jumbotron>
          <h1>Sorry this page does not exists...</h1>
        </Jumbotron>
      </div>
    );
  }
  // #endregion
}

export default PageNotFound;
