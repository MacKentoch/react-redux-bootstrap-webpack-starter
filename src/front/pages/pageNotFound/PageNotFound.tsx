import React from 'react';
import { RouteComponentProps } from 'react-router';
import Jumbotron from 'reactstrap/lib/Jumbotron';
import FadeInEntrance from '../../components/fadeInEntrance';

// #region types
type Props = {} & RouteComponentProps;
// #endregion

function PageNotFound(props: Props) {
  return (
    <FadeInEntrance>
      <Jumbotron>
        <h1>Sorry this page does not exists...</h1>
      </Jumbotron>
    </FadeInEntrance>
  );
}

PageNotFound.displayName = 'PageNotFound';

export default PageNotFound;
