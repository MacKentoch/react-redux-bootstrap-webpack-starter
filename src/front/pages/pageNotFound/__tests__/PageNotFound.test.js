// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import PageNotFound from '../PageNotFound';
// #endregion

describe('PageNotFound page', () => {
  it('renders as expected', () => {
    const props = {
      actions: {},
    };
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <PageNotFound {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
