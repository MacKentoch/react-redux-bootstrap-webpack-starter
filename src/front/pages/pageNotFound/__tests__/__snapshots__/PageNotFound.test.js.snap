// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound page renders as expected 1`] = `
<div>
  <div>
    <div
      className="jumbotron"
    >
      <h1>
        Sorry this page does not exists...
      </h1>
    </div>
  </div>
</div>
`;
